### PR TITLE
DrawingView DisconnectHandler

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

https://github.com/dotnet/maui/issues/3604

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs [hunk 1]
```diff
@@ -12,6 +12,20 @@ public class DrawingView : View, IDrawingView
...
+	public DrawingView()
+	{
+		Unloaded += DrawingView_Unloaded;
+	}
+
+	void DrawingView_Unloaded(object? sender, EventArgs e)
+	{
+		Unloaded -= DrawingView_Unloaded;
+		Handler?.DisconnectHandler();
+	}
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs:27]: Thanks Vlad! Let's move this method down next to the other private methods in the file.

**Review by brminnick — CHANGES_REQUESTED** [review]
Thanks Vlad! Looks good 👍 

The change requested is just a cosmetic request to keep the private methods together in the file.

File: src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs [hunk 2]
```diff
@@ -175,4 +183,10 @@ void IDrawingView.DrawingLineCompleted(IDrawingLine lastDrawingLine)
 			DrawingLineCompletedCommand.Execute(lastDrawingLine);
 		}
 	}
-}
\ No newline at end of file
+
+	void DrawingView_Unloaded(object? sender, EventArgs e)
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs:187]: just a name change
```suggestion
	void OnDrawingViewUnloaded(object? sender, EventArgs e)
```

File: src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs [hunk 3]
```diff
@@ -12,6 +12,14 @@ public class DrawingView : View, IDrawingView
 {
 	readonly WeakEventManager drawingLineCompletedEventManager = new();
 
+	/// <summary>
+	/// Initializes a new instance of the <see cref="DrawingView"/> class.
+	/// </summary>
+	public DrawingView()
+	{
+		Unloaded += DrawingView_Unloaded;
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs:20]:
```suggestion
		Unloaded += OnDrawingViewUnloaded;
```

File: src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs [hunk 4]
```diff
@@ -175,4 +183,10 @@ void IDrawingView.DrawingLineCompleted(IDrawingLine lastDrawingLine)
 			DrawingLineCompletedCommand.Execute(lastDrawingLine);
 		}
 	}
-}
\ No newline at end of file
+
+	void DrawingView_Unloaded(object? sender, EventArgs e)
+	{
+		Unloaded -= DrawingView_Unloaded;
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Views/DrawingView/DrawingView.shared.cs:189]:
```suggestion
		Unloaded -= OnDrawingViewUnloaded;
```

**Review by pictos — APPROVED** [review]
LGTM!

**Review by brminnick — APPROVED** [review]

**Pull request merged**